### PR TITLE
Speed-up CI

### PR DESCRIPTION
**Pull request by sttts**
- split test workflow into test and test-e2e
- limit concurrency to 2 for e2e

e2e tests with this run in 5:30min, instead of 7:50min before.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,10 +98,11 @@ imports: $(OPENSHIFT_GOIMPORTS)
 	$(OPENSHIFT_GOIMPORTS) -m github.com/kcp-dev/kcp
 
 COUNT ?= 5
+E2E_PARALLELISM ?= 1
 
 .PHONY: test-e2e
 test-e2e: install
-	go test -tags e2e -race -count $(COUNT) ./test/e2e... $(WHAT)
+	go test -tags e2e -race -count $(COUNT) -p $(E2E_PARALLELISM) ./test/e2e... $(WHAT)
```
Comment (ncdc) [inline Makefile:105]: Do you want to set `-parallel` too?

> -parallel n
    Allow parallel execution of test functions that call t.Parallel.
    The value of this flag is the maximum number of tests to run
    simultaneously; by default, it is set to the value of GOMAXPROCS.
    Note that -parallel only applies within a single test binary.
    The 'go test' command may run tests for different packages
    in parallel as well, according to the setting of the -p flag
    (see 'go help build').

Comment (sttts) [reply Makefile:105]: that's `-p`, no?

Impressive how bad the go help is. I don't see `-p` there. Go the hint from some googling. So I assume it is `-parallel`.

Comment (ncdc) [reply Makefile:105]: There's both `-p` and `-parallel`. `-p` is documented in `go help build`.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -78,7 +78,18 @@ jobs:
...
+  e2e:
+    name: e2e
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: v1.17
+      - run: make build
+      - run: ARTIFACT_DIR=/tmp/e2e PATH="${PATH}:$(pwd)/bin/" COUNT=1 E2E_PARALLELISM=2 make test-e2e
```
Comment (stevekuznetsov) [inline .github/workflows/ci.yaml:92]: Why COUNT=1 ? NACK - we've already found flakes with COUNT=5

File: Makefile (hunk 1 above)
Comment (sttts) [reply Makefile:105]:
```
-p n
		the number of programs, such as build commands or
		test binaries, that can be run in parallel.
		The default is GOMAXPROCS, normally the number of CPUs available.
```
versus
-parallel n
	    Allow parallel execution of test functions that call t.Parallel, and
	    fuzz targets that call t.Parallel when running the seed corpus.
	    The value of this flag is the maximum number of tests to run
	    simultaneously.
	    While fuzzing, the value of this flag is the maximum number of
	    subprocesses that may call the fuzz function simultaneously, regardless of
	    whether T.Parallel is called.
	    By default, -parallel is set to the value of GOMAXPROCS.
	    Setting -parallel to values higher than GOMAXPROCS may cause degraded
	    performance due to CPU contention, especially when fuzzing.
	    Note that -parallel only applies within a single test binary.
	    The 'go test' command may run tests for different packages
	    in parallel as well, according to the setting of the -p flag
	    (see 'go help build').
```
So yes, `--parallel` needs to be set too. Wondering what applies then, the product of both?

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (sttts) [reply .github/workflows/ci.yaml:92]: running tests 5 times?

Can we at least split the e2e into one that does COUNT=1 and returns early, and another with COUNT=5 ?

I hate to wait.

Comment (sttts) [reply .github/workflows/ci.yaml:92]: e2e and e2e-quick

Comment (stevekuznetsov) [reply .github/workflows/ci.yaml:92]: How long does `-count 5` take and what is the comparison to unit tests? If it makes you wait more than 5 extra minutes, I'm more keen to try fewer counts...

Comment (sttts) [reply .github/workflows/ci.yaml:108]: Added benchmark commit. Let's see.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**